### PR TITLE
Fixes #31686 - Highlight changed matchers

### PR DESCRIPTION
**Pull request by xprazak2**
I might need some UX guidance here on how to best highlight the changed fields and what colour to use. Background works good for input fields, but if we want to do background for checkbox we would need a custom one. I tried outlines as well, but they do not look that great. Any suggestions?

input-highlights

### CODE REVIEW

**Comment by ares** [comment]
personally I like it, I asked Maria from UXD to also take a look

**Comment by MariSvirik** [comment]
Firstly, I must say that there’s been a shift to PF4 as the main design system across Red Hat products - and that's my field of expertise.
Right now we don’t really have a pattern to display changed fields. Although we can display incorrect input like this:  
Screenshot 2021-01-22 at 12 03 55
But adding an explanatory text under the field would change the height of the field.

Screenshot 2021-01-22 at 12 03 08

Very frequently it's this done by highlighting the background of elements that need attention. (As you did. Maybe use lighter yellow or less opacity) 
But if you opt for any option (background or outlined) you should explain to the users what it means. e.g by using inline notification on the top of the page or small icons with a tooltip next/in the changed field.

One more thing you should take a look at is also spacing in the table. Fields in the columns are toooo close to each other to deliver a legible experience.
Nice work nevertheless.

**Comment by xprazak2** [comment]
Thank you for suggestions @MariSvirik!

I decreased the colour opacity of the background and added icons with hints:
icons-pop

**Comment by ares** [comment]
LGTM, @xprazak2 once this is in, could you pretty please open the same against foreman_ansible (ansible variables) and https://github.com/theforeman/foreman_puppet? If you open new redmine issues, please also link them to https://bugzilla.redhat.com/show_bug.cgi?id=1926840

File: app/assets/stylesheets/puppetclasses.scss
Comment (ezr-ondrej) [inline app/assets/stylesheets/puppetclasses.scss]: can we please use a different css file ? I'd rather see a new one specific for `lookup_keys`, this one is going to get removed with puppet, but this addition should stay in core.

File: app/assets/javascripts/lookup_keys.js
Comment (ezr-ondrej) [inline app/assets/javascripts/lookup_keys.js]: could we add these new functions to webpacked js on `window.tfm` object in some subkey?
We will probably need new file, but I think we should not use the global functions anymore.
Something like `webpack/assets/javascripts/foreman_lookup_keys.js` would do nicely IMHO.

**Review by ezr-ondrej — COMMENTED** [review]
Just some comments about the code placement, otherwise looks good enough to me.

**Comment by ezr-ondrej** [comment]
> could you pretty please open the same against foreman_ansible (ansible variables) and foreman_puppet

No need, this will be used by both of those as they both use these templates, because the matchers are generic :+1:

**Comment by xprazak2** [comment]
I moved the new js into webpack and styles into a new stylesheet file.

**Comment by ezr-ondrej** [comment]
We have some linting issues now, but apart of that looks good to me now :+1:

**Review by ezr-ondrej — APPROVED** [review]
Works for me :+1:
Thanks @xprazak2!

**Pull request merged**
